qiskit/quantum_info/mitigation/complete_readout_mitigator.py

```diff
@@ -0,0 +1,275 @@
+        else:
+            qubits = tuple(sorted(qubits))
+
+        # Check for cached mitigation matrix
+        # if not present compute
+        if qubits not in self._mitigation_mats:
+            marginal_matrix = self.assignment_matrix(qubits)
+            try:
+                mit_mat = np.linalg.inv(marginal_matrix)
+            except np.linalg.LinAlgError:
+                # Use pseudo-inverse if matrix is singular
+                mit_mat = np.linalg.pinv(marginal_matrix)
+            self._mitigation_mats[qubits] = mit_mat
+
+        return self._mitigation_mats[qubits]
+
+    def assignment_matrix(self, qubits: List[int] = None) -> np.ndarray:
+        r"""Return the readout assignment matrix for specified qubits.
+        The assignment matrix is the stochastic matrix :math:`A` which assigns
+        a noisy readout probability distribution to an ideal input
+        readout distribution: :math:`P(i|j) = \langle i | A | j \rangle`.
+        Args:
+            qubits: Optional, qubits being measured.
+        Returns:
+            np.ndarray: the assignment matrix A.
+        """
+        if qubits is None:
+            return self._assignment_mat
+
+        if isinstance(qubits, int):
+            qubits = [qubits]
+
+        # Compute marginal matrix
+        axis = tuple(
+            self._num_qubits - 1 - i for i in set(range(self._num_qubits)).difference(qubits)
+        )
+        num_qubits = len(qubits)
+        new_amat = np.zeros(2 * [2 ** num_qubits], dtype=float)
+        for i, col in enumerate(self._assignment_mat.T[self._keep_indexes(qubits)]):
+            new_amat[i] = (
+                np.reshape(col, self._num_qubits * [2]).sum(axis=axis).reshape([2 ** num_qubits])
+            )
+        new_amat = new_amat.T
+        return new_amat
+
+    @staticmethod
+    def _keep_indexes(qubits):
+        indexes = [0]
+        for i in sorted(qubits):
+            indexes += [idx + (1 << i) for idx in indexes]
+        return indexes
+
+    @staticmethod
+    def _z_diagonal(dim, dtype=float):
+        r"""Return the diagonal for the operator :math:`Z^\otimes n`"""
+        parity = np.zeros(dim, dtype=dtype)
+        for i in range(dim):
+            parity[i] = bin(i)[2:].count("1")
+        return (-1) ** np.mod(parity, 2)
+
+    @staticmethod
+    def _expval_with_stddev(
+        coeffs: np.ndarray, probs: np.ndarray, shots: int
+    ) -> Tuple[float, float]:
+        """Compute expectation value and standard deviation.
+        Args:
+            coeffs: array of diagonal operator coefficients.
+            probs: array of measurement probabilities.
+            shots: total number of shots to obtain probabilities.
+        Returns:
+            tuple: (expval, stddev) expectation value and standard deviation.
+        """
+        # Compute expval
+        expval = coeffs.dot(probs)
+
+        # Compute variance
+        sq_expval = (coeffs ** 2).dot(probs)
+        variance = (sq_expval - expval ** 2) / shots
```

Comment:
I think this method under-estimates error since it doesn't take into account the mitigation overhead from the CTMP paper. In that paper they use an upper bound for the variance based on the A-matrix which comes from uncertainty in the estimates of the true error probabilities.

Reply:
Does the CTMP mitigation overhead estimator work for all mitigation methods? If so - perhaps we should add `mitigation_overhead` to the base class?